crates/pico_core/src/index.rs

```diff
@@ -0,0 +1,21 @@
+use std::fmt;
+
+use crate::epoch::Epoch;
+
+#[derive(Clone, Copy)]
+pub struct Index {
```

Comment:
nit: can `Index` be generic over the type of some phantom data? Seeing `DashMap<ParamId, Index>` makes me nervous